aten/src/ATen/core/Tensor.h

```diff
@@ -324,6 +324,28 @@ class CAFFE2_API Tensor {
       bool keep_graph = false,
       bool create_graph = false);
 
+  /// Increments the version count of this tensor.
+  void bump_version() noexcept {
+    unsafeGetTensorImpl()->bump_version();
+  }
+
+  /// Sets this tensor's version counter.
+  void set_version_counter(
+      const c10::VariableVersion& version_counter) noexcept {
+    unsafeGetTensorImpl()->set_version_counter(version_counter);
+  }
+
+  /// Retrieves this tensor's version counter.
+  const c10::VariableVersion& version_counter() const noexcept {
+    return unsafeGetTensorImpl()->version_counter();
+  }
+
+  /// Retrieves the current value of this tensor's version counter.
+  /// Equivalent to calling `version_counter().current_version()`.
+  uint32_t current_version() const noexcept {
+    return unsafeGetTensorImpl()->version_counter().current_version();
+  }
```

Comment:
Seeing these on the public API makes me a little unhappy, because we don't have them on `torch.Tensor`, but I guess they were on `Variable` before so this is the easiest thing to do for now.